build_cmake/CMakeLists.txt

```diff
@@ -2,17 +2,39 @@ cmake_minimum_required(VERSION 2.8)
 project(CodeGen)
 
 set(CMAKE_MODULE_PATH
-	${CodeGen_SOURCE_DIR}/Modules
+	${CodeGen_SOURCE_DIR}/../../Dependencies/cmake-modules
 	${CMAKE_MODULE_PATH}
 )
 
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release CACHE STRING
+		"Choose the type of build, options are: None Debug Release"
+		FORCE)
+endif()
+
+if(NOT MSVC)
+	if(CMAKE_BUILD_TYPE STREQUAL "Release")
+		add_definitions(-DNDEBUG)
+	else()
+		add_definitions(-D_DEBUG)
+	endif()
+endif()
+
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 if (NOT TARGET Framework)
 	add_subdirectory(../../Framework/build_unix ${CodeGen_SOURCE_DIR}/../../Framework/build_unix/build)
```

Comment:
This is outdated and should be updated to reflect the changes in the Framework repository.